### ISSUE TITLE
WYSIWYG QEditor buttons on Safari 13/FF reset cursor to wrong position + others

### ISSUE THREAD

**Issue by cesboa** [quasarframework/quasar#5277]
When changing font (eg. bold, italic, underline) in the "Editor (WYSIWYG)" using Safari (Version 13.0.1 (14608.2.11.1.11)) the text cursor resets to to start of the input field and doesn't change the text. It works as expected in Chrome (Version 77.0.3865.120 (Official Build) (64-bit)).

https://codepen.io/pen/?&editable=true&editors=101

1. Mark text with cursor, eg. "you see is"
2. Click on bold
3. Text cursor is now at start of the line, text hasn't changed.

Quasar version: 1.2.0
OS: MacOS 10.14.6
Browsers: Safari Version 13.0.1 (14608.2.11.1.11)

**Comment by cesboa** [comment]
Any news on this, it's still an issue several updates later.

**Comment by Kirimoar** [comment]
I'd be interested in this as well - it seems like the complete QEditor functions where you mark something before action do not work correctly, which is a major problem as most of the MAC-Users use Safari.

**Comment by cesboa** [comment]
So, this is just dead in the water now. The WYSIWYG editor is useless on with safari on ios and macos... I don't understand, many major updates to quasar has been released, but this rather huge bug has not been handled in nearly two months.

**Comment by rstoenescu** [comment]
Fix will be available in "quasar" v1.5.8

**Comment by RobertoLangarica** [comment]
**Here is a fix in case someone is stuck with an older version of Quasar.**

The problem is because the tags used in the QEditor are not buttons, and Safari seems to dislike that, so the workaround is to tweek a little bit the properties used by QEditor where the tag <a> is hardcoded and change it for a <button>

> Note: The workaround was tested with quasar v1.1.2 y Safari v13.0.4.

The next code was in a boot script, but it will work in any other place. Basically when the platform is Safari we are replacing  the previous button type from anchor to button, so when the toolbar gets created it will use buttons instead of anchors.


```javascript
import {QEditor, Platform} from 'quasar'

function fixedButtonProps() {
  const flat = this.toolbarOutline !== true &&
          this.toolbarPush !== true

  return {
    type: 'button',
    flat,
    noWrap: true,
    outline: this.toolbarOutline,
    push: this.toolbarPush,
    rounded: this.toolbarRounded,
    dense: true,
    color: this.toolbarColor,
    disable: !this.editable,
    size: 'sm'
  }
}

// "async" is optional
export default async ({ Vue }) => {
  // Change only for Safari
  if (Platform.is.safari) {
    QEditor.options.computed.buttonProps = fixedButtonProps
  }
}
```
